### ISSUE TITLE
about mean_delta in batchnorm_layer.c

### ISSUE THREAD

**Issue by lixiaoqing123456** [AlexeyAB/darknet#4605]
Hi, the code in batchnorm_layer.c to calculate mean_delta
```
void mean_delta_cpu(float *delta, float *variance, int batch, int filters, int spatial, float *mean_delta)
{

    int i,j,k;
    for(i = 0; i < filters; ++i){
        mean_delta[i] = 0;
        for (j = 0; j < batch; ++j) {
            for (k = 0; k < spatial; ++k) {
                int index = j*filters*spatial + i*spatial + k;
                mean_delta[i] += delta[index];
            }
        }
        mean_delta[i] *= (-1./sqrt(variance[i] + .00001f));
    }
}
```
seems missing a part of the gradient.
According to the paper,  https://arxiv.org/abs/1502.03167, 
![image](https://user-images.githubusercontent.com/26628768/71643066-4682bb00-2cef-11ea-8588-0af85d7aa018.png)
Does the missing part have little effect on the training of the network?
